ansible/undercloud/deploy-undercloud.yml

```diff
@@ -0,0 +1,200 @@
+        owner: stack
+        group: stack
+
+    - name: Import instackenv.json
+      shell: . /home/stack/stackrc; openstack baremetal import --json /home/stack/instackenv.json
+
+    - name: Configure boot
+      shell: . /home/stack/stackrc; openstack baremetal configure boot
+
+    - name: Introspect Nodes
+      shell: . /home/stack/stackrc; openstack baremetal introspection bulk start
+
+    - name: Remove boot_option:local from ironic nodes
+      shell: . /home/stack/stackrc; ironic node-list | grep None | awk '{print $2}' | xargs -I % ironic node-update % replace properties/capabilities='cpu_vt:true,cpu_hugepages:true,cpu_txt:true,cpu_aes:true,cpu_hugepages_1g:true'
+
+    - name: Remove boot_option:local from flavors
+      shell: ". /home/stack/stackrc; openstack flavor unset --property 'capabilities:boot_option' {{item}}"
+      with_items:
+        - baremetal
+        - compute
+        - control
+
+    - name: Copy original network configuration files
+      shell: cp -r /usr/share/openstack-tripleo-heat-templates/network/config/single-nic-vlans ~/templates/nic-configs
+
+    - name: Copy over static configuration files
+      copy:
+        src: "{{item.src}}"
+        dest: "{{item.dest}}"
+        owner: stack
+        group: stack
+      with_items:
+        - src: files/compute.yaml
+          dest: /home/stack/templates/nic-configs/compute.yaml
+        - src: files/controller.yaml
+          dest: /home/stack/templates/nic-configs/controller.yaml
+        - src: files/tunings.yaml
+          dest: /home/stack/tunings.yaml
+
+    - name: Copy network-environment.yaml
+      template:
+        src: "templates/network-environment.yaml.j2"
+        dest: /home/stack/templates/network-environment.yaml
+        owner: stack
+        group: stack
+
+    # (WARNING) This is more than likely a hack to allow external network addresses on compute nodes
+    - name: Fix /usr/share/openstack-tripleo-heat-templates/environments/network-isolation.yaml
+      become: true
+      lineinfile:
+        dest: /usr/share/openstack-tripleo-heat-templates/environments/network-isolation.yaml
+        line: "  OS::TripleO::Compute::Ports::ExternalPort: ../network/ports/external.yaml"
+        regexp: "  OS::TripleO::Compute::Ports::ExternalPort: ../network/ports/noop.yaml"
+        backup: true
+
+    # TODO: Fix what is causing ansible to complain of a syntax issue with this debug statement
+    # - debug: msg="You should be ready to deploy: date;time openstack overcloud deploy --templates -e /usr/share/openstack-tripleo-heat-templates/environments/network-isolation.yaml -e /home/stack/templates/network-environment.yaml -e /usr/share/openstack-tripleo-heat-templates/environments/puppet-pacemaker.yaml -e /home/stack/tunings.yaml  --libvirt-type=kvm --neutron-public-interface nic1 --ntp-server clock.corp.redhat.com --neutron-network-type vxlan --neutron-tunnel-types vxlan --control-scale 3 --compute-scale 13  ;date"
```

Comment:
Can you do a global sed -i 's/redhat.com/example.com' . on this please?  We want to remove any internal references to stuff.